dev/tools/svg2dart/lib/svg2dart.dart

```diff
@@ -0,0 +1,204 @@
+    if (childElement.name.local == 'path') {
+      paths.add(SvgPath.fromElement(childElement));
+    }
+  }
+  return new FrameData(new Point<double>(width, height), paths);
+}
+
+// Given a points list in the form e.g: "25.0, 1.0 12.0, 12.0 23.0, 9.0" matches
+// the coordinated of the first point and the rest of the string, for the
+// example above:
+// group 1 will match "25.0"
+// group 2 will match "1.0"
+// group 3 will match "12.0, 12.0 23.0, 9.0"
+final RegExp _pointMatcher = new RegExp(r' *([\-\.0-9]+) *, *([\-\.0-9]+)(.*)');
+
+/// Parse a string with a list of points, e.g:
+/// '25.0, 1.0 12.0, 12.0 23.0, 9.0' will be parsed to:
+/// [Point(25.0, 1.0), Point(12.0, 12.0), Point(23.0, 9.0)].
+List<Point<double>> parsePoints(String points) {
+  String unParsed = points;
+  List<Point<double>> result = [];
+  while(unParsed.isNotEmpty && _pointMatcher.hasMatch(unParsed)) {
+    Match m = _pointMatcher.firstMatch(unParsed);
+    result.add(new Point(
+        double.parse(m.group(1)),
+        double.parse(m.group(2))
+    ));
+    unParsed = m.group(3);
+  }
+  return result;
+}
+
+/// Data for a single animation frame.
+class FrameData {
+  const FrameData(this.size, this.paths);
+
+  final Point<double> size;
+  final List<SvgPath> paths;
+
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+          other is FrameData &&
+              runtimeType == other.runtimeType &&
+              size == other.size &&
+              const ListEquality<SvgPath>().equals(paths, other.paths);
+
+  @override
+  int get hashCode =>
+      size.hashCode ^
+      paths.hashCode;
+
+  @override
+  String toString() {
+    return 'FrameData{size: $size, paths: $paths}';
+  }
+}
+
+/// Represents an SVG path element.
+class SvgPath {
+  const SvgPath(this.id, this.commands);
+
+  final String id;
+  final List<SvgPathCommand> commands;
+
+  static final RegExp _pathCommandMatcher = new RegExp(r'([MmLlQqCcAZz]) *([\-\.0-9 ,]*)');
+
+  static SvgPath fromElement(XmlElement pathElement) {
+    assert(pathElement.name.local == 'path');
+    String id = _extractAttr(pathElement, 'id');
+    String dAttr = _extractAttr(pathElement, 'd');
+    List<SvgPathCommand> commands = <SvgPathCommand> [];
+    for (Match match in _pathCommandMatcher.allMatches(dAttr)) {
+      String commandType = match.group(1);
+      String pointStr = match.group(2);
+      commands.add(new SvgPathCommand(commandType, parsePoints(pointStr)));
+    }
+    return new SvgPath(id, commands);
+  }
+
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+          other is SvgPath &&
+              runtimeType == other.runtimeType &&
+              id == other.id &&
+              const ListEquality<SvgPathCommand>().equals(commands, other.commands);
+
+  @override
+  int get hashCode =>
+      id.hashCode ^
+      commands.hashCode;
+
+  @override
+  String toString() {
+    return 'SvgPath{id: $id, commands: $commands}';
```

Comment:
we normally use (...)

Reply:
Change (note that this was auto generated by intellij).